### PR TITLE
remove all instances of `@at-root`

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

Either removed `@at-root` completely (was redundant) or replaced with `:is` where needed.

This is in preparation of scoping v2 classes, because scoping will rely on nesting, and `@at-root` breaks nesting.

## Testing

All tests should pass.

## Docs

N/A

### CODE REVIEW

File: packages/itwinui-css/src/input/input.scss [hunk 1]
```diff
@@ -48,7 +48,7 @@
     cursor: not-allowed;
   }
 
-  @at-root :where(textarea#{&}) {
+  &:where(textarea) {
```
Comment (rohan-kadkol) [inline packages/itwinui-css/src/input/input.scss:51]: Specificity goes up with this. Should we have something like this to keep the old specificity?

```suggestion
  :where(textarea:is(&))
```

```css
/* Old */
:where(textarea.iui-input) {} /* 0,0,0 */
:where(textarea.iui-input)[disabled], :where(textarea.iui-input)[readonly] {} /* 0,1,0 each */

/* New */
.iui-input:where(textarea) {} /* 0,1,0 */
.iui-input:where(textarea)[disabled], .iui-input:where(textarea)[readonly] {} /* 0,2,0 each */
```

File: packages/itwinui-css/src/table/table.scss [hunk 2]
```diff
@@ -146,10 +146,11 @@
     border-radius: inherit;
   }
 
-  @at-root .iui-table-header & {
+  .iui-table-header & {
     flex-grow: 1;
   }
-  @at-root .iui-table-body & {
+
+  .iui-table-body & {
```
Comment (rohan-kadkol) [inline packages/itwinui-css/src/table/table.scss:153]: nit: Unnecessary formatting change

File: packages/itwinui-css/src/input/input.scss (hunk 1 above)
Comment (mayank99) [reply packages/itwinui-css/src/input/input.scss:51]: i think it doesn't really matter

File: packages/itwinui-css/src/table/table.scss (hunk 2 above)
Comment (mayank99) [reply packages/itwinui-css/src/table/table.scss:153]: where? `@at-root` was removed from here.

**Review by rohan-kadkol — APPROVED** [review]

**Review by gretanausedaite — APPROVED** [review]

**Review by FlyersPh9 — APPROVED** [review]
